internal/loadbalancer/resource_network.go

```diff
@@ -2,358 +2,531 @@ package loadbalancer
+	resp.Diagnostics.Append(newDiags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *NetworkResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema.MarkdownDescription = util.MarkdownDescription(`
+Manage the attachment of a Load Balancer in a Network in the Hetzner Cloud.
+`)
+
+	resp.Schema.Attributes = map[string]schema.Attribute{
+		"id": schema.StringAttribute{
+			Computed: true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
 		},
-		Schema: map[string]*schema.Schema{
-			"network_id": {
-				Type:         schema.TypeInt,
-				ExactlyOneOf: []string{"network_id", "subnet_id"},
-				Optional:     true,
-				ForceNew:     true,
+		"load_balancer_id": schema.Int64Attribute{
+			MarkdownDescription: "ID of the Load Balancer.",
+			Required:            true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.RequiresReplace(),
 			},
-			"subnet_id": {
-				Type:         schema.TypeString,
-				ExactlyOneOf: []string{"network_id", "subnet_id"},
-				Optional:     true,
-				ForceNew:     true,
+		},
+		"network_id": schema.Int64Attribute{
+			MarkdownDescription: "ID of the Network to attach the Load Balancer to. Using `subnet_id` is preferred. Required if `subnet_id` is not set. If `subnet_id` or `ip` are not set, the Load Balancer will be attached to the last subnet (ordered by `ip_range`).",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.RequiresReplace(),
+				int64planmodifier.UseStateForUnknown(),
 			},
-			"load_balancer_id": {
-				Type:     schema.TypeInt,
-				Required: true,
-				ForceNew: true,
+		},
+		"subnet_id": schema.StringAttribute{
+			MarkdownDescription: "ID of the Subnet to attach the Load Balancer to. Required if `network_id` is not set.",
+			Optional:            true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
 			},
-			"ip": {
-				Type:     schema.TypeString,
-				Computed: true,
-				Optional: true,
-				ForceNew: true,
+		},
+		"ip": schema.StringAttribute{
+			CustomType:          iptypes.IPAddressType{},
+			MarkdownDescription: "IP to assign to the Load Balancer.",
+			Optional:            true,
+			Computed:            true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.RequiresReplace(),
+				stringplanmodifier.UseStateForUnknown(),
 			},
-			"enable_public_interface": {
-				Type:     schema.TypeBool,
-				Optional: true,
-				Default:  true,
+			Validators: []validator.String{
+				validateutil.IP(),
 			},
 		},
+		// XXX: Move to `load_balancer` since it is unrelated to the given private
+		// network attachment.
+		"enable_public_interface": schema.BoolAttribute{
+			MarkdownDescription: "Wether the Load Balancer public interface is enabled. Default is `true`.",
+			Optional:            true,
+			Computed:            true,
```

Comment:
Should not need `computed` with the default, as we always write it -> it always matches the config.

Reply:
Default does require the computed to be true, or the provide fails with a developer error.